src/module.js

```diff
@@ -34,74 +37,121 @@ const execute = (command, log) => {
     });
 };
 
+function escapeRegExp (string) {
+    // $& means the whole matched string
+    return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+function uuidFromVMName (vmName, uuid, log) {
+    return new Promise((resolve, reject) => {
+        if (uuid) {
+            return resolve(uuid);
+        }
+
+        log.info(`Running 'VBoxManage list vms' to locate installed VM named '${ vmName }'`);
+        execute('VBoxManage list vms', log)
```

Comment:
I wonder if the [`showvminfo`](https://www.virtualbox.org/manual/ch08.html#vboxmanage-showvminfo) command would be a better fit to retrieve the vm name. But I might be wrong. I don't know the ins and outs of the command line interface of virtual box. Did you choose the `list` command for a particular reason?

Reply:
Alright, after putting some more thought into it, I think using `showvminfo` might be a little better as it frees us from using a generated regex. However I am happy to do this after merging your pull request. I want to add the same functionality to the sister project ([karma-virtualbox-edge-launcher](https://github.com/chrisguttandin/karma-virtualbox-edge-launcher)) anyway.